migrations/Version202203021737164734_taoBackOffice.php

```diff
@@ -0,0 +1,162 @@
+        'taoQtiItem' => [
+            'qtiCreator' => [
+                'item/multiColumn' => [
+                    'multi-column' => []
+                ],
+                'item/scrollableMultiColumn' => [
+                    'scrollable-multi-column' => []
+                ],
+                'item/perInteractionRp' => [
+                    'perInteractionRp' => []
+                ],
+            ]
+        ]
+    ];
+
+    public function up(Schema $schema): void
+    {
+        $resultConfig = self::DEFAULT_CONFIG;
+        $backOfficeExtension = \common_ext_ExtensionsManager::singleton()->getExtensionById('taoBackOffice');
+        if($backOfficeExtension === null) {
+            throw new \Exception(sprintf('Cannot find %s extension', 'taoBackOffice'));
+        }
+        $existingConfig = $backOfficeExtension->getConfig(self::GLOBAL_UI_CONFIG_NAME);
+        if($existingConfig !== false) {
+            $resultConfig = $existingConfig;
+        }
+
+        foreach(self::EXISTING_CONFIG_MAP as $extensionName => $extensionConfig) {
+            $currentExtension = \common_ext_ExtensionsManager::singleton()->getExtensionById($extensionName);
+            if($currentExtension === null) {
+                throw new \Exception(sprintf('Cannot find %s extension', $extensionName));
+            }
+            foreach($extensionConfig as $configFileName => $existingConfig) {
+                $currentExtensionConfig = $currentExtension->getConfig($configFileName);
+                if($currentExtensionConfig === false) {
+                    throw new \Exception(sprintf('Cannot find %s config', $configFileName));
+                }
+
+                $this->updateGlobalUIConfig($existingConfig, $currentExtensionConfig,$resultConfig);
+            }
+        }
+
+        $backOfficeExtension->setConfig(self::GLOBAL_UI_CONFIG_NAME, $resultConfig);
+    }
+
+    private function updateGlobalUIConfig($existingConfig, $currentExtConfig, array &$resultConfig): void
+    {
+        foreach($existingConfig as $globalConfigKey => $configData) {
+            foreach($configData as $configKey => $singleConfig) {
+                if(!array_key_exists($configKey, $currentExtConfig)) {
+                    continue;
+                }
+                if (!empty($singleConfig)) {
+                    foreach($singleConfig as $value) {
+                        if(!array_key_exists($value, $currentExtConfig[$configKey])) {
+                            continue;
+                        }
+                        $resultConfig[$globalConfigKey] = $currentExtConfig[$configKey][$value] ? self::SHOW_OPTION : self::HIDE_OPTION;
+                    }
+                } else {
+                    $resultConfig[$globalConfigKey] = $currentExtConfig[$configKey] ? self::SHOW_OPTION : self::HIDE_OPTION;
+                }
+            }
+        }
+    }
+
+    public function down(Schema $schema): void
+    {
+        $this->throwIrreversibleMigrationException(
```

Comment:
I'm not so sure it's not reversible. We're not changing any existing configuration that could cause a breaking change. Do we really need that?

Reply:
You mean the `down` method? Because it is part of interface and we need to implement it.